arkouda/pdarrayclass.py

```diff
@@ -725,64 +724,122 @@ def __getitem__(self, key):
-            if key < 0:
-                # Interpret negative key as offset from end of array
-                key += self.size
-            if key >= 0 and key < self.size:
-                generic_msg(
-                    cmd="[int]=val",
-                    args={
-                        "array": self,
-                        "idx": key,
-                        "dtype": self.dtype,
-                        "value": self.format_other(value),
-                    },
-                )
-            else:
-                raise IndexError(f"index {orig_key} is out of bounds with size {self.size}")
-        elif isinstance(key, pdarray):
-            if isinstance(value, pdarray):
-                generic_msg(cmd="[pdarray]=pdarray", args={"array": self, "idx": key, "value": value})
+        if self.ndim == 1:
+            if np.isscalar(key) and (resolve_scalar_dtype(key) in ["int64", "uint64"]):
+                orig_key = key
+                if key < 0:
+                    # Interpret negative key as offset from end of array
+                    key += self.size
+                if key >= 0 and key < self.size:
+                    generic_msg(
+                        cmd="[int]=val-1D",
+                        args={
+                            "array": self,
+                            "idx": key,
+                            "dtype": self.dtype,
+                            "value": self.format_other(value),
+                        },
+                    )
+                else:
+                    raise IndexError(f"index {orig_key} is out of bounds with size {self.size}")
+            elif isinstance(key, pdarray):
+                if isinstance(value, pdarray):
+                    generic_msg(
+                        cmd="[pdarray]=pdarray", args={"array": self, "idx": key, "value": value}
+                    )
+                else:
+                    generic_msg(
+                        cmd="[pdarray]=val",
+                        args={
+                            "array": self,
+                            "idx": key,
+                            "dtype": self.dtype,
+                            "value": self.format_other(value),
+                        },
+                    )
+            elif isinstance(key, slice):
+                (start, stop, stride) = key.indices(self.size)
+                logger.debug(f"start: {start} stop: {stop} stride: {stride}")
+                if isinstance(value, pdarray):
+                    generic_msg(
+                        cmd="[slice]=pdarray",
+                        args={
+                            "array": self,
+                            "start": start,
+                            "stop": stop,
+                            "stride": stride,
+                            "value": value,
+                        },
+                    )
+                else:
+                    generic_msg(
+                        cmd="[slice]=val-1D",
```

Comment:
I am a little confused by the use of the `1D`s here. Why is it only on some of them and why is it hardcoded to 1D rather than doing the f string based on number of dims?

Reply:
The `1D` is only on the `[int]=val` and `[slice]=val` commands because I didn't update the other two to support ND arrays yet — that will probably come in a future PR because I don't think it's needed for the array API. It's hard-coded to one here because this code branch is only for 1D arrays. There is more code further down that uses the array's rank in the command.